src/core/server/elasticsearch/integration_tests/client.test.ts

```diff
@@ -55,3 +63,67 @@ describe('elasticsearch clients', () => {
     expect(resp.headers!.warning).toMatch('system indices');
   });
 });
+
+function createFakeElasticsearchServer() {
+  const firstRequestReceived$ = new Subject<boolean>();
+
+  const requestHandler = jest.fn((req, res) => {
+    // Reply with a 200 and empty response by default (intentionally malformed response)
+    res.writeHead(200);
+    res.end();
+
+    firstRequestReceived$.next(true);
+  });
+
+  const server = http.createServer(requestHandler);
+  server.listen(esTestConfig.getPort());
+  return { server, requestHandler, firstRequestReceived$ };
+}
+
+describe('fake elasticsearch', () => {
+  let esServer: http.Server;
+  let requestHandler: jest.Mock;
+  let kibanaServer: Root;
+
+  beforeAll(async () => {
+    kibanaServer = createRootWithCorePlugins(
+      {
+        logging: { silent: false },
+        status: { allowAnonymous: true },
+      },
+      { verbose: true }
+    );
+    const fakeServer = await createFakeElasticsearchServer();
+    esServer = fakeServer.server;
+    requestHandler = fakeServer.requestHandler;
+
+    await kibanaServer.preboot();
+    await kibanaServer.setup();
+    kibanaServer.start();
+    await fakeServer.firstRequestReceived$.pipe(first()).toPromise(); // Waiting for the first ES request instead of the start to complete, because start can't never complete due to waiting for ES to be ready
+  });
+
+  afterAll(async () => {
+    await kibanaServer.shutdown();
+    await esServer.close();
+  });
+
+  test('should return unknown product when it cannot perform the Product check (503 response)', async () => {
+    expect(requestHandler).toHaveBeenCalledWith(
+      expect.objectContaining({
+        method: 'GET',
```

Comment:
Isn't it an implementation detail? Why we should adjust the test whenever an URL is modified?

Reply:
You're right, we can remove this validation :)

Reply:
Removed 👍